src/jit/emitxarch.cpp

```diff
@@ -3033,7 +3033,6 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
                     offset = 0;
 
                     // Ensure that all the GenTreeIndir values are set to their defaults.
-                    assert(memBase->gtRegNum == REG_NA);
```

Comment:
Assert was incorrect. Under JitStress=1 on [Windows x64 Checked](https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_jitstress1_prtest/42/) we had a few tests fail.

Validated that we still want to go the `emit_R_S` or `emit_S_R` route by checking the code-diff.